#! /usr/bin/env python
# Task Long configuration

PACKAGE='floor_nav'
import roslib; roslib.load_manifest(PACKAGE)
import math

from dynamic_reconfigure.parameter_generator import *
from task_manager_lib.parameter_generator import *

gen = TaskParameterGenerator()
#       Name                    Type            Description
#       Default    Min   Max
gen.add("front_sector",  bool_t, 0,    "If true, use a front sector",  False)
gen.add("angular_range",  double_t, 0,    "Size of angular sensors (in rad)", 30.*math.pi/180., 0.0, math.pi )
gen.add("safety_range",  double_t, 0,    "Distance at which we need to stop", 0.25, 0.0, 5.0 )
gen.add("dont_care_range",  double_t, 0,    "Distance at which we can ignore obstacles", 1.0, 0.0, 5.0 )
gen.add("max_linear_speed",  double_t, 0,    "Max allowed linear speed", 1.00, 0.0, 5.0 )
gen.add("max_angular_speed",  double_t, 0,    "Max allowed angular speed", 1.00, 0.0, 5.0 )


exit(gen.generate(PACKAGE, "floor_nav", "TaskWander"))

